python_modules/dagster/dagster/core/log_manager.py

```diff
@@ -148,12 +148,91 @@ def construct_log_string(
     )
 
 
+def get_dagster_meta_dict(
+    logging_metadata: DagsterLoggingMetadata, dagster_message_props: DagsterMessageProps
+) -> Dict[str, Any]:
+    # combine all dagster meta information into a single dictionary
+    meta_dict = {
+        **logging_metadata._asdict(),
+        **dagster_message_props._asdict(),
+    }
+    # step-level events can be logged from a pipeline context. for these cases, pull the step
+    # key from the underlying DagsterEvent
+    if meta_dict["step_key"] is None:
+        meta_dict["step_key"] = dagster_message_props.step_key
+
+    return meta_dict
+
+
+class DagsterLogConversionHandler(logging.Handler):
```

Comment:
I think the convention for naming handlers is to name them by where they send the logs.  E.g. StreamHandler sends logs to a stream, FileHandler sends logs to a file, and NullHandler sends logs... nowhere.  I'm not attached to any particular name, and I agree that the naming convention is a little funky, for the reason you pointed out, but I think the naming convention is worth following.

https://docs.python.org/3/library/logging.handlers.html